ddtrace/contrib/aiokafka/patch.py

```diff
@@ -0,0 +1,258 @@
+        return await func(*args, **kwargs)
+
+    topic, value, headers, partition, key, servers = parse_send(instance, args, kwargs)
+
+    with core.context_with_data(
+        "aiokafka.send",
+        parent=None,
+        span_name=schematize_messaging_operation(PRODUCE, provider="kafka", direction=SpanDirection.OUTBOUND),
+        span_type=SpanTypes.WORKER,
+        call_key="instrumented_send",
+        service=trace_utils.ext_service(pin, config.aiokafka),
+        tags=create_send_span_tags(topic, partition, key, value, servers),
+        pin=pin,
+    ) as ctx, ctx[ctx["call_key"]] as span:
+        core.dispatch("aiokafka.send.start", (topic, value, key, headers, span))
+        args, kwargs = set_argument_value(args, kwargs, 5, "headers", headers, override_unset=True)
+
+        result = await func(*args, **kwargs)
+
+        # Callback when message is actually sent
+        def sent_callback(future):
+            core.dispatch("aiokafka.send.completed", (future.result(),))
+
+        result.add_done_callback(sent_callback)
+        return result
+
+
+async def traced_send_and_wait(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    topic, value, headers, partition, key, servers = parse_send(instance, args, kwargs)
+
+    with core.context_with_data(
+        "aiokafka.send_and_wait",
+        parent=None,
+        span_name=schematize_messaging_operation(PRODUCE, provider="kafka", direction=SpanDirection.OUTBOUND),
+        span_type=SpanTypes.WORKER,
+        call_key="instrumented_send_and_wait",
+        service=trace_utils.ext_service(pin, config.aiokafka),
+        tags=create_send_span_tags(topic, partition, key, value, servers),
+        pin=pin,
+    ) as ctx, ctx[ctx["call_key"]] as span:
+        core.dispatch("aiokafka.send.start", (topic, value, key, headers, span))
+        args, kwargs = set_argument_value(args, kwargs, 5, "headers", headers, override_unset=True)
+
+        result = await func(*args, **kwargs)
+        core.dispatch("aiokafka.send.completed", (result,))
+        return result
+
+
+def convert_headers_for_distributed_tracing(aiokafka_headers):
+    dd_headers = {}
+    for header in aiokafka_headers:
+        dd_headers[header[0]] = header[1].decode("utf-8")
+    return dd_headers
+
+
+def create_get_span_tags(instance):
+    tags = {
+        COMPONENT: config.aiokafka.integration_name,
+        SPAN_KIND: SpanKind.CONSUMER,
+        HOST_LIST: bootstrap_servers(instance),
+        MESSAGING_SYSTEM: SERVICE,
+        GROUP_ID: instance._group_id,
+    }
+    return tags
+
+
+async def traced_getone(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    err = None
+    result = None
+    child_of = None
+    try:
+        result = await func(*args, **kwargs)
```

Comment:
Since our tracing setup does not allow us to change a span's parent, we have traditionally called the wrapped consume function before starting the consume span. However, we also need to record the start time of the original wrapped call, then reset the span's start time to be before the original consume. Please see how this is done for our confluent-kafka integration. [Here](https://github.com/DataDog/dd-trace-py/blob/e7786202b22cc7bfea2694dc7fb31967f03c0d29/ddtrace/contrib/internal/kafka/patch.py#L212) we capture the start time, then overwrite the spans start time [here](https://github.com/DataDog/dd-trace-py/blob/e7786202b22cc7bfea2694dc7fb31967f03c0d29/ddtrace/contrib/internal/kafka/patch.py#L248).